### PR TITLE
plugins/ocp : Fixed the OCP Telemetry String Log Page (LID:C9h) fix

### PR DESCRIPTION
**Pull request by VigneshwaranSaravana**
This PR will take care the below failures for "telemetry-string-log"
Resolved the json printing issue
Changed the Statistic Identifier, Event String, Vendor Unique, ASCII table offset and size calculation 

Signed-off-by: Vigneshwaran Saravanan/Vigneshwaran Saravanan <s.vignesh@samsung.com>

Reviewed-by: Karthik Balan <karthik.b82@samsung.com>, 
Reviewed-by:  Arunpandian J <arun.j@samsung.com>, 
Reviewed-by:  Jayakanthan Rajendran <jaya.ganthan@samsung.com>

### CODE REVIEW

**Comment by igaw** [comment]
Please address the issues reported by checkpatch

```shell
WARNING: Missing commit description - Add an appropriate one
Error: WARNING: Missing commit description - Add an appropriate one


WARNING: Unexpected content after email: 'Karthik Balan <karthik.b82@samsung.com>, Arunpandian J', should be: 'Karthik Balan <karthik.b82@samsung.com> (, Arunpandian J)'
#8:
Reviewed-by: Karthik Balan <karthik.b82@samsung.com>, Arunpandian J
Error: WARNING: Unexpected content after email: 'Karthik Balan <karthik.b82@samsung.com>, Arunpandian J', should be: 'Karthik Balan <karthik.b82@samsung.com> (, Arunpandian J)'


ERROR: space required before the open brace '{'
#61: FILE: plugins/ocp/ocp-nvme.c:2900:
+ if (log_data->sitsz != 0){
Error: ERROR: space required before the open brace '{'

[...]
```

**Comment by VigneshwaranSaravana** [comment]
@igaw - I have resolved all the coding guidelines issue. Even I have updated the commit description but still patch run failed due to WARNING: Missing commit description - Add an appropriate one. Could you please check

**Comment by igaw** [comment]
I have the same feedback as in the other PR (https://github.com/linux-nvme/nvme-cli/pull/2374#issuecomment-2194381370)

**Comment by igaw** [comment]
I don't see any changes here. Still the version from last week.

File: plugins/ocp/ocp-nvme.c [hunk 1]
```diff
@@ -3271,7 +3327,7 @@ static int get_c9_log_page(struct nvme_dev *dev, char *format)
 	int ret = 0;
 	__u8 *header_data;
 	struct telemetry_str_log_format *log_data;
-	nvme_print_flags_t fmt;
+	enum nvme_print_flags fmt;
```
Comment (igaw) [inline plugins/ocp/ocp-nvme.c:3330]: I suggest we keep this type as `nvme_print_flags_t` here. We just changed the printing flags type to this. We define the flags as enum type but then we want to be able to use several bits in variable. Thus this shouldn't be an enum anymore.

Comment (VigneshwaranSaravana) [reply plugins/ocp/ocp-nvme.c]: @igaw 
Fixed. Please check.

I have fixed all the coding guideline issue. Patch failed due to below one. Even I have update commit description. Please check
-------------------------------------------------------------
[1/1] Check commit - c143e40863a693e58db373b89134fbfa5f4efb15
-------------------------------------------------------------
WARNING: Missing commit description - Add an appropriate one
Error: WARNING: Missing commit description - Add an appropriate one

Comment (igaw) [reply plugins/ocp/ocp-nvme.c]: Your commit message needs to be edited. This has nothing to do what you edit in github's PR. checkpatch complains about the git commit itself.

`git commit --edit --amend` should open your text editor and you should fix this

image

**Comment by igaw** [comment]
@arthurshau as we are almost there with this PR, could you have a look?

**Comment by arthurshau** [comment]
Logic LGTM

**Comment by igaw** [comment]
Updated the commit message myself.

**Comment by igaw** [comment]
Thanks.

**Pull request merged**